backend/vouch/trust_algo.py

```diff
@@ -1,110 +1,128 @@
+         vouch_matrix[voucher][vouchee] > 0 if voucher vouched for vouchee.
+    pretrusts -- pretrusts[u] > 0 if u is pretrusted.
     """
-    # Size of the pre-trusted set.
-    trust_size = np.sum(np.array(trust_status) > 0, axis=0)
+    normalized_vouch_matrix = np.zeros(vouch_matrix.shape)
+
+    nb_users = len(pretrusts)                                 # Number of users
+    nb_pretrusted = np.sum(np.array(pretrusts) > 0, axis=0)   # Number of pretrusted users
 
-    for i in range(len(C[0])):
-        vouching_size = np.sum(np.array(C[i]) > 0, axis=0)
-        for j in range(len(C[0])):
-            trust_indic = int(trust_status[j] > 0)
-            vouch_indic = int(C[i][j] > 0)
-            C[i][j] = ALPHA / (ALPHA * trust_size + vouching_size) * trust_indic
-            C[i][j] += 1 / (ALPHA * trust_size + vouching_size) * vouch_indic
-    return C
+    for voucher in range(nb_users):
+        n_vouches_by_voucher = np.sum(np.array(vouch_matrix[voucher]) > 0, axis=0)
+        normalization_constant = IMPLICIT_PRETRUST_VOUCH * nb_pretrusted + n_vouches_by_voucher
+        for vouchee in range(nb_users):
+            if pretrusts[vouchee] > 0:
+                normalized_vouch_matrix[voucher][vouchee] \
+                    += IMPLICIT_PRETRUST_VOUCH / normalization_constant
+            if vouch_matrix[voucher][vouchee] > 0:
+                normalized_vouch_matrix[voucher][vouchee] += 1 / normalization_constant
+    return normalized_vouch_matrix
 
 
-def get_trust_vector(C, p):
+def compute_relative_posttrusts(normalized_vouch_matrix, relative_pretrusts: NDArray):
     """
     Return a vector of global trust values per user, given the vouchers in the
     network and the set of pre-trusted users. This part comes directly from
     EigenTrust.
     """
-    t = p
-    newt = t
+    relative_trusts = relative_pretrusts
+    new_relative_trusts = relative_trusts
     delta = 10
-    while delta >= EPS:
-        newt = C.T.dot(t)
-        newt = (1 - A) * newt + A * p
-        delta = np.linalg.norm(newt - t)
-        t = newt
-    return newt
+    while delta >= APPROXIMATION_ERROR:
+        new_relative_trusts = normalized_vouch_matrix.T.dot(relative_trusts)
+        new_relative_trusts = (1 - PRETRUST_BIAS) * new_relative_trusts \
+            + PRETRUST_BIAS * new_relative_trusts
+        delta = np.linalg.norm(new_relative_trusts - relative_trusts)
+        relative_trusts = new_relative_trusts
+    return new_relative_trusts
 
 
-def rescale(trust_vector, trust_status):
+def compute_voting_rights(relative_posttrusts, pretrusts):
     """
     Go from ratio of trust to actual voting weight that should be assigned to
     users given the trust the network puts in them.
     """
-    mint = np.amin(
-        [trust_vector[i] for i in range(len(trust_vector)) if trust_status[i] > 0]
+    min_relative_trusts_of_pretrusteds = np.amin(
+        [relative_posttrusts[u] for u in range(len(relative_posttrusts)) if pretrusts[u] > 0]
     )
```

Comment:
Similarly, this can be simplified with [`np.amin(where=...)`](https://numpy.org/doc/stable/reference/generated/numpy.amin.html#numpy.amin). 

```suggestion
    min_relative_trusts_of_pretrusteds = np.amin(
        relative_posttrusts,
        where=pretrusts > 0,
        initial=MIN_PRETRUST_VOTING_RIGHT,
    )
```

And the "initial" value prevents `np.amin()` from failing if no "pretrusted" user exists.